### PR TITLE
MWPW-129397 - Bulk updating links on iternational customer stories 

### PR DESCRIPTION
**Pull request by JasonHowellSlavin**
- Adds code to fetch urls from query index
- Adds code to update links in international customer stories 
- Adds bare bones structure for Bacom code

### CODE REVIEW

File: bacom/fix-breadcrumb-link.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    return entry.split('-')[0];
+}
+
+/**
+ * A variation of the "getLeaf" function. Removes the second conditional check for
+ * "!node.children", which was returning the incorrect item
+ * 
+ * 
+ * @param {object} node / mdast json object with children
+ * @param {stirng} type / a type, like "link"
```
Comment (meganthecoder) [inline bacom/fix-breadcrumb-link.js:38]:
```suggestion
 * @param {string} type / a type, like "link"
```

File: bacom/fix-breadcrumb-link.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ * @returns 
+ */
+const getLink = (node, type) => {
+    if (node?.type === type) return node;
+  
+    if (node.children) {
+      for (let i = 0; i < node.children.length; i++) {
+        const leaf = getLink(node.children[i], type);
```
Comment (meganthecoder) [inline bacom/fix-breadcrumb-link.js:46]: If it has a child then it's not a leaf node, right? Should the variable name be changed?

File: bacom/fix-breadcrumb-link.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        const mdast = await getMdast(text);
+        const breadcrumbs = getTable(mdast, 'breadcrumbs');
+        const homeLink = getLink(breadcrumbs[0].table, 'link');
+        let linkToChange;
+
+        if (homeLink) linkToChange = homeLink.url;
+
+        console.log('Looking at link', homeLink, linkToChange, linkToChange == incorrectURL);
+
+        if (homeLink && homeLink.url === incorrectURL) {
```
Comment (meganthecoder) [inline bacom/fix-breadcrumb-link.js:88]:
```suggestion
        if (homeLink?.url === incorrectURL) {
```

File: index.js [hunk 4]
```diff
@@ -16,7 +16,10 @@ import { mdast2docx } from '@adobe/helix-md2docx';
 import parseMarkdown from '@adobe/helix-html-pipeline/src/steps/parse-markdown.js';
 
 const entries = [
-    'https://main--bacom--adobecom.hlx.page/drafts/cmillar/ben-and-jerrys-case-study',
+    'https://main--bacom-blog--adobecom.hlx.page/how-to/10-secrets-to-delivering-a-great-product-demo',
+    'https://main--bacom-blog--adobecom.hlx.page/how-to/10-easy-steps-to-connecting-over-the-top-ott-devices-to-a-proxy',
+    'https://main--bacom-blog--adobecom.hlx.page/how-to/12-social-media-tactics-drive-traffic-your-ecommerce-site',
+    'https://main--bacom-blog--adobecom.hlx.page/how-to/3-best-practices-to-transform-to-a-data-driven-operating-model',
```
Comment (meganthecoder) [inline index.js:22]: What's the purpose of these changes?

**Review by meganthecoder — APPROVED** [review]
Nice work! I added a few cleanup items but overall looks good. I'm not very familiar with mdast updates though. Perhaps @sukamat could review?

**Comment by JasonHowellSlavin** [comment]
No longer needed, outdated methods.
